### ISSUE TITLE
[$500] Profile - Time Zone is not updated on secondary device after changing on main device

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#18036]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go https://staging.new.expensify.com/
and login on 2 devices
2. Change Time Zone in profile on main device
3. Verify Time Zone on secondary device

## Expected Result:
Time Zone should be updated

## Actual Result:
Time Zone is not updated

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome
- [ ] MacOS / Desktop

**Version Number:** 1.3.6.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/234680900-adb1ef08-98ba-4a9b-97b9-6d3c35dd0945.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ad9119a009eaf79a</li>
        <li>Upwork Job ID: 1651357597669793792</li>
        <li>Last Price Increase: 2023-10-11</li>
<li>Automatic offers: </li>
<ul>
<li>0xmiroslav | Contributor | 26911215</li>
</ul></li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @conorpendergrast (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by marcaaron** [comment]
I don't know why we are expecting this to work in realtime, but it would require a backend change and therefore is an `Internal` issue. Gonna take a quick look at this and will put it back where I found it if not immediately clear...

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01ad9119a009eaf79a

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor Plus for review of internal employee PR - @0xmiroslav (`Internal`)

**Comment by marcaaron** [comment]
This works fine for me.

**Comment by marcaaron** [comment]
@kbecciv did we try testing on some other platform combinations to see if this is unique to Android & Windows?

**Comment by kbecciv** [comment]
@marcaaron I can request the QA team to reproduce the issue with different devices. Let me know and I will do that.

**Comment by marcaaron** [comment]
Yes please, let's reopen this if we can reproduce it. I am guessing this was some kind of temporary disruption because I can't reliably reproduce.

**Comment by conorpendergrast** [comment]
I could reliably produce this, when changing on my OSX native client (1.3.15-0) and iOS client app (1.3.15-5). To compare, I also changed pronouns between devices (always by changing the OSX app setting first). 

Changing pronouns: 20 seconds, precisely, three times
Changing the _Automatically determine your location_ setting: within less than a second
Changing timezone: I gave up after 2 minutes, it just didn't change on the mobile app (even after I exited and re-entered the Profile page)

So, yeah, reproduced!

**Comment by conorpendergrast** [comment]
@marcaaron This is internal, correct?

**Comment by marcaaron** [comment]
yeah, I am going to guess that this is related to some of the issues on staging with Pusher and multiple events stuff. I don't really have a great handle on what exactly is happening there and I'm going OOO next week so I'd probably suggest that we re-add the engineering label and get someone else to help if this is urgent.

FWIW I think the 3 week warning about WAQ is a bit unreasonable since it's only been "reopened" for 2 days.

**Comment by conorpendergrast** [comment]
I think it's worth making progress on this, this week, so I re-added the label!

**Comment by conorpendergrast** [comment]
I think we don't need a C+ here either, so unassigned @0xmiroslav

**Comment by marcaaron** [comment]
Thanks @conorpendergrast, gonna re-add myself so I can catch up when I get back.

**Comment by conorpendergrast** [comment]
@marcochavezf Have you had any time to take a look at this one?

**Comment by conorpendergrast** [comment]
FYI, this was closed for around 2 weeks, so we probably have a little more leeway but we should still get this moving

**Comment by marcochavezf** [comment]
Sorry, I've been focused on anonymous accounts that should be live next week.

**Comment by conorpendergrast** [comment]
Waiting on @marcochavezf, who is prioritising anonymous accounts

**Comment by marcaaron** [comment]
I'm back from OOO now - but this issue is not a priority for me. Having user settings update in realtime on multiple devices is pretty "edge case". So, I am going to sit on it until I have cycles to investigate. If someone feels this is more urgent then feel free to take over.

**Comment by marcaaron** [comment]
Going to drop this to weekly. I have higher priority things to look at.

**Comment by marcaaron** [comment]
Not overdue. Still just a lower priority bug and not something that people are going to do frequently or notice is broken IMO.

**Comment by rezkiy37** [comment]
Hi, I'm Michael (Mykhailo) from Callstack and I would like to work in this issue.

**Comment by conorpendergrast** [comment]
Hey Michael (Mykhailo)! As this is an internal issue we're only able to have internal engineers contribute, I'm sorry to say! The label `Internal` versus `External` and `Help Wanted` are the key signifiers here. Thank you!

**Comment by marcaaron** [comment]
I've got a few things to check out in NewDot today so I will try to prioritize this and get it ❌ out or at least put some kind of summary together for why it's happening.

**Comment by marcaaron** [comment]
I'm sorry everyone, but again, **I can't reproduce this**. 

https://github.com/Expensify/App/assets/32969087/8748fd14-6562-4851-8ddc-6e16c831ff5f

Without a reproduction there is not much that is actionable for me to do here.

Some possibilities come to mind...

- Pusher connection has failed and realtime features no longer work for some reason
- API issue caused the update to not happen for some reason
- Connectivity issue on the user's end caused an update to be missed for some reason

Without a specific and recent example to investigate logs and look into I don't feel like there is anything meaningful that I can achieve with this assignment.

If there is some issue with Pusher then the changes we are planning to introduce as part of the **[Reliable Notifications](https://github.com/Expensify/Expensify/issues/292523)** and **[Improve Pusher](https://github.com/Expensify/Expensify/issues/288587)** project should help to a certain degree. 

We may never have a "perfect realtime" app for a variety of reasons. But we are working on making things as stable as we can.

@conorpendergrast how do you want to proceed?

**Comment by conorpendergrast** [comment]
Shoot, sorry, I missed this last week. 

That's so weird, when I re-tested today, between my real New Expensify account on iOS and OSX, it doesn't ever seem to update the time zone. That's not _realtime_ or not, I mean that the timezone never appears to update on the second device

**Comment by conorpendergrast** [comment]
eg, this is manager.pendergrast@expensicorp.com signed in Staging in Chrome and Safari Incognito. In this fascinating 40 second video, you see me changing to manually set timezone in Safari, and then changing the time zone in Safari, and nothing changes in Chrome 

https://github.com/Expensify/App/assets/5802477/9d4a15fe-130d-4eeb-a79a-089122c48d9a

**Comment by conorpendergrast** [comment]
Guh ok, there has to be something else going on here. Nothing I do in this account (manager.pendergrast@expensicorp.com) in Safari Incognito, is updating in Chrome. 

eg 1: pronouns


https://github.com/Expensify/App/assets/5802477/d4d8e0be-363d-4a58-a179-327b0fce58ea

eg 2: Profile name: 

https://github.com/Expensify/App/assets/5802477/b48aa33b-95ac-4054-a9fe-d98ca6d9d485

**Comment by conorpendergrast** [comment]
Right, I tried and brand new account (conor+18036AA@conorpendergrast.com) and that worked like absolute magic across Chrome and Safari Private: 

https://github.com/Expensify/App/assets/5802477/3706c1db-53c2-4dca-a94e-bb1a014ec002

**Comment by marcaaron** [comment]
yep, works fine for me. A bit slow maybe. But updates fine.

I did check logs though and it looks like you are getting some kind of network error on iOS at least

https://www.expensify.com/_devportal/tools/logSearch/#sort=asc&size=10000&query=blob%3A%20%22UpdateSelectedTimezone%22%20AND%20email%3A%22manager.pendergrast%40expensicorp.com%22%20AND%20timestamp%3A%5B2023-07-14T00%3A00%20TO%202023-07-15T23%3A59%5D

Unfortunately, the errors are also really opaque so I can't give you many options for how to resolve that.

Have you tried completely logging out of those devices, clearing all stored data, and logging back in?

**Comment by marcaaron** [comment]
This log is also interesting...

https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%227e6a1cd47cc63622-MAN%22)+AND+timestamp:[2023-07-14T12:29:06.134Z+TO+2023-07-14T14:29:06.134Z]&index=logs_expensify-023302

It shows that you are updating like one zillions of people when you change your timezone so that is probably why you are experiencing this as "broken". I bet it's just slow.

Curious to get some thoughts from @yuwenmemon on this. I can't really think of how to make this more performant off the top of my head, but we can probably prioritizing updating the requesting account first.

**Comment by marcaaron** [comment]
Hmm ok so, that didn't work. I have an account locally with about 4000 personal details and something about it is borking. I got this error locally:

```
[alrt] Push_API - Failed sending parallel pusher event Pusher\ApiErrorException - 659502eeccc0ff081a5078da3800d059 ~~ exceptionMessage: 'cURL error 6: getaddrinfo() thread failed to start#012 (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://api-mt1.pusher.com/apps/792137/batch_events?auth_key=ac6d22b891daae55283a&auth_timestamp=1689384679&auth_version=1.0&body_md5=08b76b1e4c8a96a7e8d4041b2b8e1df4&auth_signature=1029dabe8e675dd66d229514d77303c9302ed4f95b020b4ad65ab4aa36fed149' exceptionFile: '/vagrant/Web-Expensify/vendor/pusher/pusher-php-server/src/Pusher.php' exceptionLine: '846' exceptionCode: '0'
```

**Comment by marcaaron** [comment]
Hmm ok so, that didn't work. I have an account locally with about 4000 personal details and something about it is borking. I got this error locally:

```
[alrt] Push_API - Failed sending parallel pusher event Pusher\ApiErrorException - 659502eeccc0ff081a5078da3800d059 ~~ exceptionMessage: 'cURL error 6: getaddrinfo() thread failed to start#012 (see https://curl.haxx.se/libcurl/c/libcurl-errors.html)
```

**Comment by marcaaron** [comment]
I am not too familiar with the parallel pusher event stuff. @iwiznia do you have any ideas?

**Comment by iwiznia** [comment]
Seems like a network problem since we failed to resolve DNS... I assume that's just one instance though and not reproducible, right?

**Comment by yuwenmemon** [comment]
> we can probably prioritizing updating the requesting account first. That way it will at least "look fast" for the person who is taking the action.

That makes sense to me :+1: - updating the timezone for everyone else is not as important for the "immediate" factor.

**Comment by marcaaron** [comment]
>I assume that's just one instance though and not reproducible, right?

@iwiznia I checked logs and it looked sort of like it has happened but triggered by contributors so I think maybe not widespread at all or only affecting dev (maybe Pusher is rate limiting the dev version of the app).

>updating the timezone for everyone else is not as important

@yuwenmemon Yeah, unfortunately - that didn't work (at least not locally) but we can try to test it out on production and see if it helps? I'll send a PR with the small change.

**Comment by marcaaron** [comment]
It will happen when it happens.

**Comment by conorpendergrast** [comment]
@isabelastisser I'm on parental leave; re-assigning! @marcaaron is working on the PR to fix this, and I don't think you'll need to do anything at all here as the bug reported was Applaused and it's 100% internal

**Comment by marcaaron** [comment]
Did time heal this bug? I think it should be fixed. Unless there is something causing it to time out. Reliable updates should also help with weird edge cases like this.

**Comment by isabelastisser** [comment]
@marcaaron, does it mean that you will work on this? 😃

**Comment by marcaaron** [comment]
It means _maybe_ **nobody** needs to work on it anymore. Reliable updates should solve mostly everything related to Pusher updates not working. There's not really any leads to investigate further so I can't justify prioritizing it as a `Daily`.

I am in favor of closing it to avoid further distraction and reopening if we run into it again after Reliable updates is enabled. 

Curious to get @danieldoglas' thoughts though. This looks really similar to some other issues related to updating the timezone timing out.

**Comment by danieldoglas** [comment]
Yeah, this should actually be testable by applause now and work. Even if they don't receive the update at that same moment, once they do a new action, we'll update the state.

**Comment by isabelastisser** [comment]
Thanks for the context, @marcaaron and @danieldoglas! 

@kbecciv, can you please see the comments above and post an update after you test it? Thanks!

**Comment by kbecciv** [comment]
Issue is reproduced in latest build 1.3.71.5

https://github.com/Expensify/App/assets/93399543/7c068aa4-18ae-413e-8d99-3bab125dd30d

**Comment by isabelastisser** [comment]
@marcaaron @danieldoglas, this is still reproducible. Should we start working on this?

**Comment by danieldoglas** [comment]
I think we can make this external and let people understand what's happening. Reliable updates are up and should work normally for applause. if this still doesn't work, I'm not against investigating it

**Comment by isabelastisser** [comment]
Great, thanks @0xmiroslav !

**Comment by marcaaron** [comment]
Waiting for proposals

**Comment by marcaaron** [comment]
@0xmiroslav please leave updates on any daily issues you are assigned to avoid them going overdue

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by isabelastisser** [comment]
@marcaaron It looks like the issue can no longer be reproduced, so I will close it now. Please feel free to reopen it if you disagree; thanks!
